diff -uNdr libopensync-plugin-gnokii-0.21.old/src/gnokii_calendar.c libopensync-plugin-gnokii-0.21/src/gnokii_calendar.c
--- libopensync-plugin-gnokii-0.21.old/src/gnokii_calendar.c	2006-08-22 20:14:15.000000000 +0200
+++ libopensync-plugin-gnokii-0.21/src/gnokii_calendar.c	2007-03-24 18:44:46.000000000 +0100
@@ -363,6 +363,7 @@
 			break;
 
 		OSyncChange *change = osync_change_new();
+		osync_change_set_member(change, env->member);
 
 		// prepare UID with gnokii-calendar-<memory location>
 		uid = g_strdup_printf ("gnokii-calendar-%i", calnote->location);
@@ -466,6 +467,7 @@
 
 			// fake a delete change to remove the old hash
 			OSyncChange *delete_change = osync_change_new();
+			osync_change_set_member(change, env->member);
 
 			// the old uid will be set for this "fake" change
 			osync_change_set_uid(delete_change, osync_change_get_uid(change));
diff -uNdr libopensync-plugin-gnokii-0.21.old/src/gnokii_contact.c libopensync-plugin-gnokii-0.21/src/gnokii_contact.c
--- libopensync-plugin-gnokii-0.21.old/src/gnokii_contact.c	2006-08-22 20:14:15.000000000 +0200
+++ libopensync-plugin-gnokii-0.21/src/gnokii_contact.c	2007-03-24 18:44:46.000000000 +0100
@@ -417,6 +417,7 @@
 				continue;
 
 			OSyncChange *change = osync_change_new();
+			osync_change_set_member(change, env->member);
 
 			// prepare UID with gnokii-contact-<memory type>-<memory location>
 			uid = gnokii_contact_uid(contact);
